python/lsst/ts/externalscripts/auxtel/calsys_take_narrowband_data.py

```diff
@@ -23,42 +17,137 @@
+        title: CalSysTakeNarrowbandData v1
+        description: Configuration for CalSysTakeNarrowbandData.
+        type: object
+        properties:
+          wavelengths:
+            type: array
+            items:
+              type: number
+              minItems: 1
+          integration_times:
+            type: array
+            items:
+              type: number
+              minItems: 1
+          fiber_spectrograph_integration_times:
+            type: array
+            items:
+              type: number
+              minItems: 1
+          mono_grating_types:
+            type: integer
+            minimum: 1
+            maximum: 3
+            default: [1]
+          mono_entrance_slit_widths:
+            type: array
+            items:
+              type: number
+              minItems: 1
+            default: [2]
+          mono_exit_slit_widths:
+            type: array
+            items:
+              type: number
+              minItems: 1
+            default: [4]
+          image_types:
+            type: array
+            items:
+              type: string
+              minItems: 1
+            default: ["Kilos"]
+          lamps:
+            type: array
+            items:
+              type: string
+              minItems: 1
+            default: ["lamps"]
+          fiber_spectrometer_delays:
+            type: array
+            items:
+              type: number
+              minItems: 1
+            default: [1]
+          latiss_filter:
+            type: array
+            items:
+              type: number
+              minItems: 1
+            default: [0]
+          latiss_grating:
+            type: array
+            items:
+              type: integer
+              minItems: 1
+            default: [0]
+          latiss_stage_pos:
+            type: array
+            items:
+              type: number
+              minItems: 1
+            default: [60]
+          nimages_per_wavelength:
+            type: array
+            items:
+              type: integer
+              minItems: 1
+            default: [1]
+          shutter:
+            type: array
+            items:
+              type: number
+              minItems: 1
+            default: [1]
+          image_sequence_name:
+            type: array
+            items:
+              type: string
+              minItems: 1
+            default: ["test"]
+          take_image:
+            type: boolean
+            default: true
+          setup_spectrograph:
+            type: boolean
+            default: true
+          file_location:
+            type: string
+            default: "~/develop"
+          script_type:
+            type: string
+            default: "narrowband"
+        required: [wavelengths, integration_times, fiber_spectrograph_integration_times]
+        additionalProperties: false
+        """
+        return yaml_schema
```

Comment:
The schema must be returned as a `dict`. See ts_standardscripts calsys_takedata.py for an example of how to do this.

Reply:
Also you are replacing a configure call that allowed scalars for values with a schema that does not.

I see plusses and minuses to this approach. On the plus side:
- The schema is significantly simpler.
- The input rules for users are simpler: always specify an array.
On the down side:
- The configuration is a bit messier for users because they have to specify arrays.
- It may feel a bit odder to have the code turn [x] into [x, x, ...x] as opposed to turning a scalar x into [x, x, ...x]

It would be a big win if all the cal scripts worked the same way, so could you please talk to @tribeiro about this. As the architect I'd prefer he sign off on it. (I am happy to join the conversation) and decide what you prefer want.

ts_standardscripts calsys_takedata.py uses the more flexible schema and would be easy to copy for this script. On the other hand if you end up decide you prefer the regularity of always specifying an array for every item then I'm happy to simplify the standard script to match.

I am happy to join the conversation if wanted.